src/persistence/Elsa.Persistence.EntityFrameworkCore/DbContexts/ElsaContext.cs

```diff
@@ -12,6 +15,11 @@ namespace Elsa.Persistence.EntityFrameworkCore.DbContexts
     public class ElsaContext : DbContext
     {
         private readonly JsonSerializerSettings serializerSettings;
+        /// <summary>
+        /// The CustomSchemaModelCacheKeyFactory will not resolve services from the DI container for constructor injection
+        /// so this is necessary in order to set the custom schema for the Model Cache.
+        /// </summary>
+        internal IDbContextCustomSchema DbContextCustomSchema { get; set; }
```

Comment:
I see. So that's why you get the custom options extension in `OnConfiguring`, which you then need to store in this internal field because you need it later in `OnModelCreating`. Seems kind of an odd design from the EF Core guys. 

But it looks like we could get the extensions from the injected `options` object in the constructor, e.g.

```
DbContextCustomSchema = options.FindExtension<CustomSchemaOptionsExtension>();
```

If you haven't done so already, maybe give that a try?

Reply:
This definitely works so I'll get rid of the OnConfiguring override and push the assignment to the constructor.

Reply:
I wish there was a better way to resolve the custom IModelCacheKeyFactory type so that it could participate in the DI flow.  I could be missing something, but I haven't found a solution just yet.